File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/context/fight/GameFightOptionToggleMessage.scala





















// Generated on 05/03/2013 10:34:40
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait GameFightOptionToggleMessage extends MessageDeserializer {
  val messageId = 707

  protected trait Message extends BaseMessage with Serializable {
    val messageId = GameFightOptionToggleMessage.messageId

    val option: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(option)
    }

  }

  protected def create(option: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val option = buf.readByte()

    create(option)
  }

}

object GameFightOptionToggleMessage extends GameFightOptionToggleMessage {
  case class MessageImpl(option: Byte) extends Message
  type Target = MessageImpl

  override def create(option: Byte) = MessageImpl(option)

  def apply(option: Byte) = create(option)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/GameFightOptionToggleMessage.java


// Generated on 05/08/2013 19:37:44
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class GameFightOptionToggleMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 707;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte option;
    
    public GameFightOptionToggleMessage() { }
    
    public GameFightOptionToggleMessage(byte option) {
        this.option = option;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(option);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        option = buf.readByte();
        if (option < 0)
            throw new RuntimeException("Forbidden value on option = " + option + ", it doesn't respect the following condition : option < 0");
    }
    
}
